Sources/SKSwiftPMWorkspace/SwiftPMWorkspace.swift

```diff
@@ -104,19 +104,19 @@ public final class SwiftPMWorkspace {
         location.workingDirectory = customWorkingDirectory
     }
 
-      // since we are customizing the workspace location, we need to explicitly pass the mirrors configuration
-    let mirrorsConfiguration = try Workspace.Configuration.Mirrors(
-        forRootPackage: packageRoot,
-        sharedMirrorFile: location.sharedMirrorsConfigurationFile,
-        fileSystem: fileSystem
-    )
+    let configuration = WorkspaceConfiguration(
+      skipDependenciesUpdates: true,
+      prefetchBasedOnResolvedFile: WorkspaceConfiguration.default.prefetchBasedOnResolvedFile,
+      additionalFileRules: WorkspaceConfiguration.default.additionalFileRules,
+      sharedDependenciesCacheEnabled: WorkspaceConfiguration.default.sharedDependenciesCacheEnabled,
+      fingerprintCheckingMode: .warn)
 
     self.workspace = try Workspace(
         fileSystem: fileSystem,
         location: location,
-        mirrors: mirrorsConfiguration.mirrors,
-        customManifestLoader: ManifestLoader(toolchain: toolchain.configuration, cacheDir: location.workingDirectory),
-        resolverUpdateEnabled: false
+        configuration: configuration,
+        customHostToolchain: toolchain,
+        customManifestLoader: ManifestLoader(toolchain: toolchain.configuration, cacheDir: location.workingDirectory)
```

Comment:
I wonder if we still need a custom manifest loader now that we are providing a custom toolchain.

ManifestLoader has only two argument: the toolchain (which we now provide explicitly) and the cacheDir. I think in the case of LSP we can use the default SwiftPM cache dir so that would mean we can just drop this argument.